spec/facility_spec.rb

```diff
@@ -23,4 +37,97 @@
       expect(@facility.services).to eq(['New Drivers License', 'Renew Drivers License', 'Vehicle Registration'])
     end
   end
-end
+
+  describe '#register vehicle' do
+    it 'can register vehicle' do
+      expect(@facility_1.registered_vehicles).to eq([])
+      expect(@facility_1.collected_fees).to eq(0)
+      expect(@cruz.registration_date).to eq (nil)
+
+      @facility_1.add_service('New Drivers License')
+      @facility_1.add_service('Renew Drivers License')
+      @facility_1.add_service('Vehicle Registration')
+      expect(@facility_1.services).to eq(['New Drivers License', 'Renew Drivers License', 'Vehicle Registration'])
+      @facility_1.register_vehicle(@cruz)
+      
+      expect(@cruz.registration_date).to eq(Date.today)
+      expect(@cruz.plate_type).to eq(:regular)
+      expect(@facility_1.collected_fees). to eq(100)
+      expect(@facility_1.registered_vehicles).to eq([@cruz])
+
+      @facility_1.register_vehicle(@camaro)
+
+      expect(@camaro.registration_date).to eq(Date.today)
+      expect(@camaro.plate_type).to eq(:antique)
+
+      @facility_1.register_vehicle(@bolt)
+
+      expect(@bolt.registration_date).to eq(Date.today)
+      expect(@bolt.plate_type).to eq(:ev)
+      expect(@facility_1.registered_vehicles).to eq([@cruz, @camaro, @bolt])
+      expect(@facility_1.collected_fees).to eq(325)
+      expect(@facility_2.registered_vehicles).to eq([])
+      expect(@facility_2.services).to eq([])
+
+      @facility_2.register_vehicle(@bolt)
+
+      expect(@facility_2.registered_vehicles).to eq([])
+      expect(@facility_2.collected_fees).to eq(0)
+    end
+  end
+
+  describe "#Getting Driver's License" do
+    it "can administer services" do
+
+      # Written Test
+      
+      expect(@registrant_1.license_data).to eq({:written=>false, :license=>false, :renewed=>false})
+      expect(@registrant_1.permit?).to be(true)
+      expect(@facility_1.administer_written_test(@registrant_1)).to be(false)
+      expect(@registrant_1.license_data). to eq({:written=>false, :license=>false, :renewed=>false})
+
+      @facility_1.add_service('Written Test')
+
+      expect(@facility_1.services).to eq(['Written Test'])
+      expect(@facility_1.administer_written_test(@registrant_1)).to be(true)
+      expect(@registrant_1.license_data).to eq({:written=>true, :license=>false, :renewed=>false})
+      expect(@registrant_2.age).to eq(16)
+      expect(@registrant_2.permit?).to be(false)
+      expect(@facility_1.administer_written_test(@registrant_2)).to be(false)
+
+      @registrant_2.earn_permit
```

Comment:
These chunks could also be broken into separate tests -- one for taking the written test if it's possible, one for NOT taking the written test if you're too young, one for NOT taking the test if you don't have a permit, etc. Then another one for earning a permit based on requirements. Another one for taking the road test. So for each service/functionality, you write a test for the happy path, and then as needed, write some tests for the sad paths (when the registrant or facility is not capable of doing that service)